### PR TITLE
Password hardening test fix

### PR DESCRIPTION
**Pull request by davidpil2002**
### Description of PR

Summary:
In continue of password-hardening sonic-mgmt tests. Link: https://github.com/sonic-net/sonic-mgmt/pull/5503
found 2 test issues:
1. one related to internal loop from fixture when calling function from tests module, fixed by moving the functions involved in the loop to a new module named password_hardening_utils.py
2. second issue is that the test set an invalid value to the expiration attribute, fixed this test issue by setting a legal value.

Fixes # (issue) 
Issue link:
https://github.com/sonic-net/sonic-mgmt/issues/6428

### Type of change

- [X ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [X ] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [ X] 202205

#### What is the motivation for this PR?
Fix the issue https://github.com/sonic-net/sonic-mgmt/issues/6428
#### How did you do it?
explained in the summary.
#### How did you verify/test it?
Run the test with the sonic-builimage build from branch 202205
command line:
py.test /sonic-mgmt/tests/passw_hardening/test_passw_hardening.py
#### Any platform specific information?
no
#### Supported testbed topology if it's a new test case?
no

### CODE REVIEW

**Comment by ZhaohuiS** [comment]
@davidpil2002 could you please add a new fixture to address teardown issue?
 _Is it possible to disable passw-harden or clean related config whenever case passed or failed, in case of blocking access to testbed and blocking other test cases?_

**Comment by davidpil2002** [comment]
@ZhaohuiS This fixture already exist
if you look in the context file, there is a a fixture named clean_passw_policies, and this fixture is cleaning all the configuration done to the default values.
this should cover your concern.

**Comment by ZhaohuiS** [comment]
@davidpil2002 I am not sure if it's enough, because last time, when I tried to run this script, it raised command error, it configured some commands on the testbed, it reminded me to change the original password, because it's too simple, otherwise, I can't login the testbed. Could you make the clean_passw_policies more robust to handle any exception in case of blocking the access to testbed? Much appreciated!

**Comment by davidpil2002** [comment]
this is meaning that the expiration of the user admin remained 0 and the feature remained enabled as well.
Now, I modified the expiration value in the test, so it will never be 0. So I think it will be ok to approved anyway.

In addition, we can in parallel debug the connection testbed issue.
the  testbed connection issue is actually not expected because the fixture  clean_passw_policies is calling the function "set_default_passw_hardening_policies" (desc below), and the function is setting default values including disabling the feature to state="disabled".

This not happened to me, can you share a full log, to see if the clean_passw_policies was called?

`def set_default_passw_hardening_policies(duthosts, enum_rand_one_per_hwsku_hostname):
    duthost = duthosts[enum_rand_one_per_hwsku_hostname]

    passw_hardening_ob_dis = test_passw_hardening.PasswHardening(state='disabled',
    passw_hardening_ob_dis = passw_hardening_utils.PasswHardening(state='disabled',
                                            expiration='100',
                                            expiration_warning='15',
                                            history='12',
                                            len_min='8',
                                            reject_user_passw_match='true',
                                            lower_class='true',
                                            upper_class='true',
                                            digit_class="true",
                                            special_class='true')`

**Comment by ZhaohuiS** [comment]
@davidpil2002 Sure, I will verify it then give you feedback in this week.

**Review by ZhaohuiS — APPROVED** [review]
Verify on physical testbed, all testcases run to success.

File: tests/passw_hardening/passw_hardening_utils.py [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    with open(pam_file_expected, 'r') as expected_common_password_file:
+        for line in expected_common_password_file:
+            line = line.strip()
+            line = line.strip('\n')
+            line = line.strip('\t')
+            common_password_expected.append(line)
+
+    common_password_diff = [li for li in difflib.ndiff(command_password_stdout, common_password_expected) if
+                            li[0] != ' ']
+    pytest_assert(len(common_password_diff) == 0, common_password_diff)
```
Comment (qiluo-msft) [inline tests/passw_hardening/passw_hardening_utils.py:90]: >common_password_diff [](http://example.com/codeflow?start=50&length=20)

The 2nd parameter should be a string, not an array.

**Pull request merged**